File: api/schemas/supplier.py
"""Tedarikçi (Supplier) Şemaları"""

from pydantic import BaseModel, ConfigDict, Field, EmailStr
from datetime import datetime
from typing import Optional


# ============ SUPPLIER USER SCHEMAS ============


class SupplierUserCreate(BaseModel):
    """Tedarikçi firma kullanıcısı oluştur"""

    name: str
    email: EmailStr
    phone: Optional[str] = None


class SupplierUserOut(SupplierUserCreate):
    """Tedarikçi firma kullanıcısı çıktı"""

    id: int
    supplier_id: int
    is_active: bool
    email_verified: bool
    password_set: bool
    is_default: bool = False

    model_config = ConfigDict(from_attributes=True)


# ============ SUPPLIER SCHEMAS ============


class SupplierCreate(BaseModel):
    """Tedarikçi oluştur"""

    company_name: str = Field(min_length=1, max_length=255)
    company_title: Optional[str] = None
    tax_number: Optional[str] = None
    registration_number: Optional[str] = None
    phone: str
    email: EmailStr
    website: Optional[str] = None
    address: Optional[str] = None
    city: Optional[str] = None
    postal_code: Optional[str] = None
    category: Optional[str] = None  # e.g., "Yazılım", "Donanım", "Hizmet"


class SupplierUpdate(BaseModel):
    """Tedarikçi güncelle"""

    company_name: Optional[str] = None
    company_title: Optional[str] = None
    email: Optional[EmailStr] = None
    tax_number: Optional[str] = None
    registration_number: Optional[str] = None
    phone: Optional[str] = None
    website: Optional[str] = None
    address: Optional[str] = None
    city: Optional[str] = None
    postal_code: Optional[str] = None
    invoice_name: Optional[str] = None
    invoice_address: Optional[str] = None
    invoice_city: Optional[str] = None
    invoice_postal_code: Optional[str] = None
    reference_score: Optional[float] = None
    notes: Optional[str] = None
    category: Optional[str] = None
    is_verified: Optional[bool] = None
    logo_url: Optional[str] = None


class SupplierOut(SupplierCreate):
    """Tedarikçi çıktı"""

    id: int
    created_by_id: int
    reference_score: float
    notes: Optional[str] = None
    is_active: bool
    is_verified: bool
    logo_url: Optional[str] = None
    created_at: datetime
    updated_at: Optional[datetime] = None

    # Invoice fields
    invoice_name: Optional[str] = None
    invoice_address: Optional[str] = None
    invoice_city: Optional[str] = None
    invoice_postal_code: Optional[str] = None
    registration_number: Optional[str] = None

    model_config = ConfigDict(from_attributes=True)


# ============ SUPPLIER QUOTE SCHEMAS ============


class SupplierQuoteItemCreate(BaseModel):
    """Tedarikçi teklif kalemi oluştur"""

    quote_item_id: int
    unit_price: float
    total_price: float
    notes: Optional[str] = None


class SupplierQuoteItemOut(SupplierQuoteItemCreate):
    """Tedarikçi teklif kalemi çıktı"""

    id: int
    supplier_quote_id: int

    model_config = ConfigDict(from_attributes=True)


class SupplierQuoteCreate(BaseModel):
    """Tedarikçi teklifi oluştur"""

    quote_id: int
    supplier_id: int
    total_amount: float
    discount_percent: Optional[float] = None
    discount_amount: Optional[float] = None
    final_amount: float
    payment_terms: Optional[str] = None
    delivery_time: Optional[int] = None
    warranty: Optional[str] = None
    notes: Optional[str] = None


class SupplierQuoteOut(SupplierQuoteCreate):
    """Tedarikçi teklifi çıktı"""

    id: int
    supplier_user_id: Optional[int] = None
    status: str
    submitted_at: Optional[datetime] = None
    created_at: datetime
    updated_at: datetime
    items: list[SupplierQuoteItemOut] = []

    model_config = ConfigDict(from_attributes=True)

File: web/src/types/supplier.ts
// web/src/types/supplier.ts

export interface Supplier {
  id: number;
  created_by_id: number;
  source_type?: "private" | "platform_network";
  company_name: string;
  logo_url?: string;
  company_title?: string;
  tax_number?: string;
  registration_number?: string;
  phone: string;
  email: string;
  website?: string;
  address?: string;
  city?: string;
  postal_code?: string;
  invoice_name?: string;
  invoice_address?: string;
  invoice_city?: string;
  invoice_postal_code?: string;
  reference_score: number;
  notes?: string;
  category?: string;
  is_active: boolean;
  is_verified: boolean;
  created_at: string;
  updated_at?: string;
  users?: SupplierUser[];
  quotes?: SupplierQuote[];
}

export interface SupplierUser {
  id: number;
  supplier_id: number;
  name: string;
  email: string;
  phone?: string;
  is_active: boolean;
  email_verified: boolean;
  password_set: boolean;
  is_default?: boolean;
  magic_token?: string;
  magic_token_expires?: string;
  created_at: string;
  updated_at?: string;
}

export interface SupplierQuote {
  id: number;
  quote_id: number;
  supplier_id: number;
  supplier_user_id?: number;
  status: string;
  total_amount: number;
  discount_percent?: number;
  discount_amount?: number;
  final_amount: number;
  payment_terms?: string;
  delivery_time?: number;
  warranty?: string;
  submitted_at?: string;
  created_at: string;
  updated_at?: string;
}

export interface SupplierQuoteItem {
  id: number;
  supplier_quote_id: number;
  quote_item_id: number;
  unit_price: number;
  total_price: number;
  notes?: string;
}
